### PR TITLE
More changes

### PR DESCRIPTION
**Pull request by pennacap**

### CODE REVIEW

**Comment by pennacap** [comment]
The test doesn't work since I changed the craft text in navbar from `Profitable crafts` to `Profitable Crafts`

**Comment by matthias-luger** [comment]
@pennacap Could you please merge develop into this fork?

**Comment by pennacap** [comment]
yeah hold on merging

**Comment by pennacap** [comment]
Fixed the bug in my code with the colors

File: api/ApiHelper.tsx [hunk 1]
```diff
@@ -76,18 +75,12 @@ export function initAPI(returnSSRResponse: boolean = false): API {
         if (!error || !error.Message) {
             return
         }
-        toast.error(error.Message, {
-            onClick: () => {
-                if (error.Trace && window.navigator.clipboard) {
-                    window.navigator.clipboard.writeText(error.Trace)
-                }
-            }
-        })
```
Comment (matthias-luger) [inline api/ApiHelper.tsx]: Why did you remove this?
Edit: After looking through all the files, it seems you have reverted multiple changes that werer made in develop. I hope that I noticed all of them and marked them to please revert.

File: components/FilterElement/FilterElement.tsx [hunk 2]
```diff
@@ -152,7 +152,7 @@ function FilterElement(props: Props) {
                     />
                 )
             } else {
-                return <EqualFilterElement key={options.name} options={options} defaultValue={props.defaultValue} onChange={onFilterElementChange} />
```
Comment (matthias-luger) [inline components/FilterElement/FilterElement.tsx]: Why is this removed? This property is needed to have a default value for pre loaded filters (for example through a external link).

File: components/FilterElement/FilterElements/EqualFilterElement.tsx [hunk 3]
```diff
@@ -13,14 +13,23 @@ export function EqualFilterElement(props: Props) {
...
     }
 
+    function getSelectOptions(option) {
+        // special case for PET_SKIN to avoid confusion
+        if (option === 'PET_SKIN') {
+            return 'Pet Skin (unapplied)'
+        }
+
+        return convertTagToName(option)
+    }
```
Comment (matthias-luger) [inline components/FilterElement/FilterElements/EqualFilterElement.tsx]: Why is this here? This is something I moved to the formatter because it was needed in multiple places...
Please remove it from here

File: components/FilterElement/FilterElements/EqualFilterElement.tsx [hunk 4]
```diff
@@ -13,14 +13,23 @@ export function EqualFilterElement(props: Props) {
...
+
     return (
         <Typeahead
             id={props.options.name}
             style={{ display: 'block' }}
             defaultSelected={props.defaultValue ? [props.defaultValue] : undefined}
             onChange={_onChange}
             options={props.options?.options}
-            labelKey={convertTagToName}
+            labelKey={getSelectOptions}
```
Comment (matthias-luger) [inline components/FilterElement/FilterElements/EqualFilterElement.tsx]: The label key should only run the "convertTagToName" function

File: components/FilterElement/FilterElements/PlayerFilterElement.tsx [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import React, { forwardRef, useState } from 'react'
 import { AsyncTypeahead } from 'react-bootstrap-typeahead'
 import api from '../../../api/ApiHelper'
+import { parsePlayer } from '../../../utils/Parser/APIResponseParser'
```
Comment (matthias-luger) [inline components/FilterElement/FilterElements/PlayerFilterElement.tsx]: why is this added?

File: components/FlipTracking/FlipTracking.tsx [hunk 6]
```diff
@@ -157,7 +159,7 @@ export function FlipTracking(props: Props) {
                             style={{ marginRight: '5px' }}
                             loading="lazy"
                         />
-                        <span style={{whiteSpace: 'nowrap' }}>{getMinecraftColorCodedElement(trackedFlip.item.name)}</span>
+                        <span style={{ ...getStyleForTier(trackedFlip.item.tier), whiteSpace: 'nowrap' }}>{trackedFlip.item.name}</span>
```
Comment (matthias-luger) [inline components/FlipTracking/FlipTracking.tsx]: This is a similar issue as above. It was changed in develop but this would revert that.
Please remove this change.

File: components/Flipper/FlipperFilter/FlipperFilter.tsx [hunk 7]
```diff
@@ -247,7 +244,7 @@ function FlipperFilter(props: Props) {
                             )
                         }
                     />
-                    <Button onClick={onProfitCalculationButtonClick}>{isCurrentCalculationBasedOnLbin(flipCustomizeSettings) ? 'lbin' : 'Median'}</Button>
+                    <Button onClick={onProfitCalculationButtonClick}>{isCurrentCalculationBasedOnLbin(flipCustomizeSettings) ? 'Lowest Bin' : 'Median'}</Button>
```
Comment (matthias-luger) [inline components/Flipper/FlipperFilter/FlipperFilter.tsx]: Change it to "Lowest BIN" as it is always written in caps.

File: components/ItemFilter/ItemFilter.tsx [hunk 8]
```diff
@@ -220,7 +221,7 @@ function ItemFilter(props: Props) {
         let defaultValue: any = ''
         if (options && options.options[0] !== null && options.options[0] !== undefined) {
             // dont set the first option for search-selects
-            if ((hasFlag(options.type, FilterType.EQUAL) && hasFlag(options.type, FilterType.SIMPLE)) || hasFlag(options.type, FilterType.BOOLEAN)) {
+            if (hasFlag(options.type, FilterType.EQUAL) || hasFlag(options.type, FilterType.BOOLEAN)) {
```
Comment (matthias-luger) [inline components/ItemFilter/ItemFilter.tsx]: Again a change from develop that would be reverted.
Revert this change

File: components/ModDetails/ModDetails.tsx [hunk 9]
```diff
@@ -1,46 +1,47 @@
 import React from 'react'
-import { Card } from 'react-bootstrap'
-
+import { Card, Button } from 'react-bootstrap'
+import DownloadIcon from '@mui/icons-material/Download'
+import styles from './ModDetails.module.css'
 function ModDetails() {
+    console.log(styles.modDetailsFooter)
```
Comment (matthias-luger) [inline components/ModDetails/ModDetails.tsx]: remove

File: components/ModDetails/ModDetails.tsx [hunk 10]
```diff
@@ -1,46 +1,47 @@
...
     return (
         <>
-            <Card>
+            <Card style={{ marginBottom: "20px" }}>
```
Comment (matthias-luger) [inline components/ModDetails/ModDetails.tsx]: After adding a seperate css file (good), it would be nice to move all the inline styling to it

File: components/ModDetails/ModDetails.tsx [hunk 11]
```diff
@@ -1,46 +1,47 @@
...
+                            Get a list of commands with <span style={{ fontFamily: "monospace" }}>/cofl help</span> or watch the{' '}
                             <a href="https://www.youtube.com/watch?v=dv66Y_cra8U&list=PLDpPmxIcq9tAssQlyJMBlSmSg5JOpq699">YouTube playlist</a> explaining them
                         </li>
                     </ol>
                     <hr />
-                    <p>
-                        Mod downloads including source code can be found on the{' '}
-                        <a target="_blank" rel="noreferrer" href="https://github.com/coflnet/skyblockmod/releases  ">
-                            Cofl Mod Github release page
+                    <div style={{ display: 'flex', justifyContent: 'space-around', alignItems: "center" }} className={styles.modDetailsFooter}>
```
Comment (matthias-luger) [inline components/ModDetails/ModDetails.tsx]: I would still like to have the video embeded, so it is instantly obvious what kind of video we want to present and the user doesnt need to navigate away from the page.

On bigger screens there is space on the left and right where you could put the links to the releases and the general github. On smaller screens just put it below.

And please use the GitHub logo from the official source: https://github.com/logos

File: components/NavBar/NavBar.tsx [hunk 12]
```diff
@@ -158,16 +155,16 @@ function NavBar(props: Props) {
             <aside className={styles.navBar} id="navBar" onMouseEnter={onMouseMove} onMouseLeave={onMouseOut}>
                 <ProSidebar id="pro-sidebar" style={style} collapsed={isCollapsed()} hidden={isHidden()}>
                     <SidebarHeader>
-                        <div style={{ padding: '24px', fontWeight: 'bold', fontSize: '20px', letterSpacing: '1px', overflow: 'hidden', whiteSpace: 'nowrap' }}>
-                            <ExploreIcon /> {!isCollapsed() ? 'Navigation' : ''}
+                        <div style={{ padding: '24px', fontWeight: 'bold', fontSize: '20px', letterSpacing: '1px', overflow: 'hidden', whiteSpace: 'nowrap', display: 'flex', alignItems: 'center'}}>
+                            <img src='/logo512.png' alt="Logo" width={"40px"} height={"40px"} style={{translate: "-5px"}}/> {!isCollapsed() ? 'CoflNet' : ''}
```
Comment (matthias-luger) [inline components/NavBar/NavBar.tsx]: Change the title to "Coflnet"

File: components/RatChecker/RatChecker.tsx [hunk 13]
```diff
@@ -112,9 +111,17 @@ function RatChecker() {
...
+                            <Form.Control type="file" className={'form-control'} style={{display: "none"}} ref={ratFileInput} onChange={(e)=>onFileUpload((e.target as HTMLInputElement).files)}multiple accept='.jar'/>
+                            <div className={styles.dNd} onClick={(e)=>ratFileInput.current?.click()} onDrop={(e)=>{
+                                onFileUpload(e.dataTransfer.files);
+                                e.preventDefault();
+                            }} onDragOver={(e)=>e.preventDefault()}>
+                                <div className={styles['dNd-content']}>
+                                    Drag and Drop Files or click to open file
+                                </div>
+                            </div>
                         </div>
```
Comment (matthias-luger) [inline components/RatChecker/RatChecker.tsx:124]: This look nice, but I would like to have the dNd-field be not as large (height). This seems unneccessary and depending on the screen size, the user cant instantly see the result without scrolling down.
I would like to have something like this (only hardcoded example):

File: components/Ref/Ref.tsx [hunk 14]
```diff
@@ -46,7 +46,7 @@ function Ref(props: Props) {
             <hr />
             <div>
                 <Card style={{ marginBottom: '15px' }}>
-                    <Card.Header>Referral system</Card.Header>
+                    <Card.Header><h2>Referral System</h2></Card.Header>
```
Comment (matthias-luger) [inline components/Ref/Ref.tsx]: Dont put a `<h2>` tag distupting the bootstrap styling.
Use the <Card.Title> that is used on most other places and was forgotten here

File: components/Ref/Ref.tsx [hunk 15]
```diff
@@ -65,7 +65,7 @@ function Ref(props: Props) {
                                 <hr />
                             </div>
                         ) : null}
-                        <p>Share your Ref-Link with people which might find Cofnet useful.</p>
+                        <p>Share your referral Link with people which might find CoflNet useful.</p>
```
Comment (matthias-luger) [inline components/Ref/Ref.tsx]: Please use "Coflnet"

Comment (matthias-luger) [inline components/Ref/Ref.tsx]: Dosnt "Referral-Link" or "Referral Link" look better?

File: components/RatChecker/RatChecker.tsx [hunk 16]
```diff
@@ -8,15 +8,16 @@ import styles from './RatChecker.module.css'
 function RatChecker() {
     let [isChecking, setIsChecking] = useState(false)
     let [checkingResults, setCheckingResults] = useState<[string, RatCheckingResponse][]>(null)
-    let ratFileInput = useRef(null)
+    let ratFileInput = useRef<HTMLInputElement>(null)
 
-    function onFileUpload(e) {
+    function onFileUpload(files) {
+        console.log(files);
```
Comment (matthias-luger) [inline components/RatChecker/RatChecker.tsx]: remove

File: pages/auction/[auction].tsx [hunk 17]
```diff
@@ -39,7 +38,7 @@ function AuctionDetailsPage(props: Props) {
     }, [auctionUUID])
 
     useEffect(() => {
-        forceUpdate()
+        forceUpdate();
```
Comment (matthias-luger) [inline pages/auction/[auction].tsx]: ; is not needed

File: pages/auction/[auction].tsx [hunk 18]
```diff
@@ -1,26 +1,25 @@
...
-import { getCacheControlHeader } from '../../utils/CacheUtils'
 
 interface Props {
     auctionDetails: any
 }
 
 function AuctionDetailsPage(props: Props) {
     const router = useRouter()
-    let auctionUUID = router.query.auctionUUID as string
+    let auctionUUID = router.query.auction as string
```
Comment (matthias-luger) [inline pages/auction/[auction].tsx]: Any reason why you changed from "auctionUUID" to "auction"? I think auctionUUID is more descriptive to instantly know what the route contains

File: pages/flipper.tsx [hunk 19]
```diff
@@ -49,7 +49,7 @@ function Flipper(props: Props) {
             {getHeadElement(undefined, 'Free auction house item flipper for Hypixel Skyblock', undefined, ['flipper'])}
             <Container>
                 <Search />
-                <h2>Item-Flipper (WIP)</h2>
+                <h2 style={{display: "flex", alignItems: "center"}}><b>Item Flipper</b>&nbsp;<span style={{backgroundColor: "white", color: "black", padding: "4px", borderRadius: "4px",fontSize:"2vh"}}>WIP</span></h2>
```
Comment (matthias-luger) [inline pages/flipper.tsx]: Remove the `b` tag. There is no reason to make this headline look different that all the others

File: pages/player/[uuid]/flips/index.tsx [hunk 20]
```diff
@@ -103,13 +103,11 @@ export function getEmbedDescription(flipTrackingResponse: FlipTrackingResponse,
...
-
     let highestProfitFlipText = `Highest Profit Flip: ${numberWithThousandsSeperators(
-        highestProfitFlip.profit
-    )} Coins \n ${numberWithThousandsSeperators(highestProfitFlip.pricePaid)} Coins ➞ ${numberWithThousandsSeperators(
-        highestProfitFlip.soldFor
-    )} Coins (${removeMinecraftColorCoding(highestProfitFlip.item.name)})`
+        flipTrackingResponse.flips[0].profit
+    )} Coins \n ${numberWithThousandsSeperators(flipTrackingResponse.flips[0].pricePaid)} Coins ➞ ${numberWithThousandsSeperators(
+        flipTrackingResponse.flips[0].soldFor
+    )} Coins (${flipTrackingResponse.flips[0].item.name})`
```
Comment (matthias-luger) [inline pages/player/[uuid]/flips/index.tsx]: Another revert from develop. 
Please revert your change

File: utils/Formatter.tsx [hunk 21]
```diff
@@ -25,11 +25,6 @@ export function convertTagToName(itemTag?: string): string {
         return ''
     }
 
-    // special case for PET_SKIN to avoid confusion
-    if (itemTag === 'PET_SKIN') {
-        return 'Pet Skin (unapplied)'
-    }
-
     // words that should remain lowercase
```
Comment (matthias-luger) [inline utils/Formatter.tsx]: Another reverted change from develop.
Please revert your change

File: utils/Formatter.tsx [hunk 22]
```diff
@@ -297,8 +292,8 @@ export function getMinecraftColorCodedElement(text: string): JSX.Element {
         a: { fontWeight: 'normal', textDecoration: 'none', color: '#55ff55' },
         b: { fontWeight: 'normal', textDecoration: 'none', color: '#55ffff' },
         c: { fontWeight: 'normal', textDecoration: 'none', color: '#FF5555' },
-        e: { fontWeight: 'normal', textDecoration: 'none', color: '#FF55FF' },
-        d: { fontWeight: 'normal', textDecoration: 'none', color: '#FFFF55' },
+        d: { fontWeight: 'normal', textDecoration: 'none', color: '#FF55FF' },
+        e: { fontWeight: 'normal', textDecoration: 'none', color: '#FFFF55' },
```
Comment (matthias-luger) [inline utils/Formatter.tsx:301]: The color codes were reverted back to the old state (https://github.com/Coflnet/hypixel-react/pull/1053) because users reported the coloring to be wrong. I dont know if this is the only mixup, but the ones currently in develop seem to work.
I dont see a reason to not use them. (You can still order them if you want)

File: utils/Formatter.tsx [hunk 23]
```diff
@@ -323,7 +318,3 @@ export function getMinecraftColorCodedElement(text: string): JSX.Element {
 
     return <span>{elements}</span>
 }
-
-export function removeMinecraftColorCoding(text: string): string {
-    return text.replace(/§[0-9a-fk-or]/ig, "");
-}
```
Comment (matthias-luger) [inline utils/Formatter.tsx]: reverted change from develop.
Please revert your change

File: utils/RateLimitUtils.tsx [hunk 24]
```diff
@@ -22,6 +22,8 @@ export default function rateLimit(options?: Options) {
                 }
                 tokenCount[0] += 1
 
+                console.log('TokenCount: ' + tokenCount)
```
Comment (matthias-luger) [inline utils/RateLimitUtils.tsx]: referted change from develop.
Please revert your change

**Review by matthias-luger — CHANGES_REQUESTED** [review]

File: components/ModDetails/ModDetails.tsx (hunk 11 above)
Comment (pennacap) [reply components/ModDetails/ModDetails.tsx]: Will do

File: pages/auction/[auction].tsx (hunk 18 above)
Comment (pennacap) [reply pages/auction/[auction].tsx]: oh i forgot to change it back, I was doing some debugging in my code, so i tried changing the query name (obviously it didn't)

File: components/RatChecker/RatChecker.tsx (hunk 13 above)
Comment (pennacap) [reply components/RatChecker/RatChecker.tsx:124]: this may be a stupid q, but how do I view the png file, it just redirects me to https://github.com/Coflnet/hypixel-react/pull/1047/files

File: components/Ref/Ref.tsx (hunk 15 above)
Comment (pennacap) [reply components/Ref/Ref.tsx]: Referral-Link is grammatically wrong, but in my opinion Referral Link looks better

File: components/RatChecker/RatChecker.tsx (hunk 13 above)
Comment (matthias-luger) [reply components/RatChecker/RatChecker.tsx:124]: Oh, no that was my mistake. Seems like I sent the comment, before the image was uploaded.
Image: 
grafik

File: components/ModDetails/ModDetails.tsx (hunk 11 above)
Comment (pennacap) [reply components/ModDetails/ModDetails.tsx]: The problem with the logo is that it has a white background, would you be fine if I took the octocat logo, cropped it to content and made the background transparent?

File: components/RatChecker/RatChecker.tsx (hunk 13 above)
Comment (pennacap) [reply components/RatChecker/RatChecker.tsx:124]: Ok so rectangular
image
This fine?

**Comment by pennacap** [comment]
finally should be done except the one's where I've responded

File: utils/Formatter.tsx (hunk 22 above)
Comment (pennacap) [reply utils/Formatter.tsx:301]: Only those two didn't work, and I fixed it in the commits.
The colors used are more vibrant and are the same ones used in Minecraft

**Comment by pennacap** [comment]
wait, don't push it into the repo, I'm fixing a few bugs + adding eslint

**Comment by matthias-luger** [comment]
@pennacap Are you still working on stuff or should I take another look at it? If you are finished just request another review or write a comment

**Comment by pennacap** [comment]
no no take a look at it

File: components/ModDetails/ModDetails.tsx (hunk 11 above)
Comment (matthias-luger) [reply components/ModDetails/ModDetails.tsx]: The png images have a transparent background. There should be no problem?
grafik

File: components/RatChecker/RatChecker.tsx (hunk 13 above)
Comment (matthias-luger) [reply components/RatChecker/RatChecker.tsx:124]: yes

**Comment by matthias-luger** [comment]
@pennacap Seems good except the 3 task I kept open. After the mod page (including the rat checker) is finshed. This can be merged :)
Please request another review when you are finished

File: components/ModDetails/ModDetails.tsx (hunk 11 above)
Comment (pennacap) [reply components/ModDetails/ModDetails.tsx]: Oh ok will use that one

**Comment by pennacap** [comment]
done @matthias-luger! i really don't see a way to put the downloads in a sidebar and make it noticeable

**Comment by matthias-luger** [comment]
LGTM!

**Comment by matthias-luger** [comment]
@pennacap I contacted you on Discord for a potential bounty
